t/FHEM/cc1101/01_func.t

```diff
@@ -120,20 +120,13 @@ InternalTimer(time(), sub {
       @{$targetHash->{QUEUE}}=();
     };
 
-    my $todo = Test2::Todo->new(reason => 'Tests needs to be corrected');
     subtest 'CalcDataRate' => sub {
       # sub input:  57, 150
       # sub output: 5c, 7a
       plan(1);
-      is(cc1101::CalcDataRate($targetHash,qw(57 150])),array {
-          item '5c'; 
-          item '7a'; 
-          end();
-        }, q[verify return values]);
+      is(cc1101::CalcDataRate($targetHash,qw(57 150)),qw(5c 7a), q[verify return values]);
```

Comment:
Warum hast Du das angepasst?

Reply:
Ich machte das weil es Fehler brachte und so nicht. So stimmt das Ergebnis mit dem Input. Getestet auch mit anderen Werte.

Ist das Ergebnis ein Array wirklich?

Reply:
Da bin ich mir recht sicher.

Der Grund, wieso ich den array Check verwend ist, dass jedes Element geprüft werden soll.

Ändert man die Zeile z.B. auf:
`is(cc1101::CalcDataRate($targetHash,qw(57 150)),qw(5d 7a), q[verify return values]);`  würden wir erwarten, dass der Test Fehl schlägt.
Macht er aber nicht, denn das letzte Element im Array passt.

Ich drehe es wieder zurück

Reply:
Wieso erhalte ich bei der Prüfung auf das Array als Vergleich beim Test immer den letzten Wert? Wenn ich auf Liste prüfe, so erhalte ich beide Werte als Vergleich. Wo denkst du, ist der Fehler hier? Die Sub arbeitet richtig.

Reply:
@sidey79 lange habe ich gesucht und gelesen / getestet. So mit der Variante prüfen wir jeden Wert ob dies stimmt.